units/generate_tasks.go

```diff
@@ -47,11 +47,19 @@ func makeGenerateTaskJob() *generateTasksJob {
 	return j
 }
 
-func NewGenerateTasksJob(id string, ts string) amboy.Job {
+// NewGenerateTasksJob returns a job that dynamically updates the project
+// configuration based on the given task's generate.tasks configuration.
+func NewGenerateTasksJob(versionID, taskID string, ts string, useScopes bool) amboy.Job {
 	j := makeGenerateTaskJob()
-	j.TaskID = id
+	j.TaskID = taskID
 
-	j.SetID(fmt.Sprintf("%s-%s-%s", generateTasksJobName, id, ts))
+	j.SetID(fmt.Sprintf("%s-%s-%s", generateTasksJobName, taskID, ts))
+	if useScopes {
+		versionScope := fmt.Sprintf("%s.%s", generateTasksJobName, versionID)
```

Comment:
Is the prefixing the scopes with `generateTasksJobName` necessary since these jobs are in a dedicated queue group already?

Reply:
It's not necessary, but it's following the existing convention of using the job name, which is sometimes helpful to indicate the scope's purpose instead of just using a raw ID.